### PR TITLE
Implement secure boot enforcement

### PR DESCRIPTION
**Pull request by andSmv**
Depending on a parameter, varstored aborts if secure_boot_enable is activated, but no certificates are present.
This behavior was discussed in [https://github.com/xapi-project/varstored/issues/16]

### CODE REVIEW

**Comment by stormi** [comment]
It's still a draft because we need to test it in situation (with a modified XAPI to set the `secureboot-enforce` setting), but otherwise is ready for review.

File: handler.c [hunk 1]
```diff
@@ -2056,6 +2057,30 @@ void dispatch_command(uint8_t *comm_buf)
...
+
+    if (!secure_boot_enforce)
+        return true;
+
+    status = internal_get_variable(EFI_SECURE_BOOT_MODE_NAME,
+                                   sizeof(EFI_SECURE_BOOT_MODE_NAME),
+                                   &gEfiGlobalVariableGuid, &data, &data_len);
+
+    if (status != EFI_SUCCESS)
+        return false;
```
Comment (stormi) [inline handler.c:2076]: If we ever enter this condition, it might deserve a specific WARN log.

File: varstored.c [hunk 2]
```diff
@@ -532,6 +543,11 @@ varstored_initialize(domid_t domid)
         }
     }
 
+    if (!check_secure_boot()) {
+        ERR("Secure boot is required, but isn't acitvated\n");
```
Comment (stormi) [inline varstored.c:547]: Typo

```suggestion
        ERR("Secure boot is required, but isn't activated\n");
```

**Comment by stormi** [comment]
Closing. See https://github.com/xapi-project/varstored/issues/16#issuecomment-1708060797.

**Comment by andSmv** [comment]
Finally, we don't need to change anymore varstored behaviour

**Comment by stormi** [comment]
Reopening following the reopening of #16

**Comment by stormi** [comment]
Looks like I can't reopen it. We'll create another.

**Pull request closed without merging**